src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs

```diff
@@ -115,6 +115,7 @@ public virtual IEnumerable<Type> ExportedTypes
         public virtual object[] GetCustomAttributes(bool inherit) { throw NotImplemented.ByDesign; }
         public virtual object[] GetCustomAttributes(Type attributeType, bool inherit) { throw NotImplemented.ByDesign; }
 
+        [RequiresAssemblyFiles(Message = "Calls 'AssemblyName.EscapeCodeBase(string)' which is a dangerous single file pattern")]
```

Comment:
There's no need to repeat the method's name - the warning generated by the analyzer should already contain that. At least that's how RequiresUnreferencedCode works, I would assume RequiresAssemblyFiles would work the same way.

Avoid "dangerous". There should be a suggestion to how to fix the problem.